src/walletSdk/index.ts

```diff
@@ -1,18 +1,115 @@
+import StellarSdk, { Networks, Server } from "stellar-sdk";
+
 import { Anchor } from "./anchor";
+import { WalletSigner, DefaultSigner } from "./auth/WalletSigner";
+import { Stellar } from "./horizon/Stellar";
+import { NETWORK_URLS } from "./horizon/constants";
+import { Recovery } from "./recovery/Recovery";
+import { getUrlDomain } from "./util/url";
+
+// TODO - https://stellarorg.atlassian.net/browse/WAL-789?atlOrigin=eyJpIjoiZjcwNzZhOTJmNjE1NGRhNTk1NDlkOTExMTYxMDJkZmYiLCJwIjoiaiJ9
+interface ClientConfigFn {}
+interface ClientConfig {}
+class HttpClient {}
 
-export enum NETWORKS {
-  PUBLIC = "PUBLIC",
-  TESTNET = "TESTNET",
+export class Config {
+  app: ApplicationConfiguration;
+  stellar: StellarConfiguration;
+  constructor(stellarCfg, appCfg) {
+    this.stellar = stellarCfg;
+    this.app = appCfg;
+  }
 }
 
 export class Wallet {
-  network: NETWORKS = NETWORKS.TESTNET;
+  private cfg: Config;
+  private clients = [];
+
+  static TestNet = (): Wallet => {
+    return new Wallet(StellarConfiguration.TestNet());
+  };
+
+  static MainNet = (): Wallet => {
+    return new Wallet(StellarConfiguration.MainNet());
+  };
+
+  constructor(
+    stellarConfiguration: StellarConfiguration,
+    applicationConfiguration: ApplicationConfiguration = new ApplicationConfiguration()
+  ) {
+    this.cfg = new Config(stellarConfiguration, applicationConfiguration);
+  }
+
+  anchor(homeDomain: string, httpClientConfig: ClientConfigFn = null) {
+    const url =
+      homeDomain.indexOf("://") !== -1 ? homeDomain : `https://${homeDomain}`;
+    return new Anchor(
+      this.cfg,
+      getUrlDomain(url),
+      this.getClient(httpClientConfig)
+    );
+  }
+
+  stellar() {
+    return new Stellar(this.cfg);
+  }
 
-  constructor(network) {
+  recover(servers, httpClientConfig) {
+    return new Recovery(
+      this.cfg,
+      this.stellar(),
+      this.getClient(httpClientConfig),
+      servers
+    );
+  }
+
+  close() {
```

Comment:
Do you need to close clients explicitly in JS?

Reply:
we shouldn;t need to explicitly close the httpclient for the JS version

Reply:
agreed, removed https://github.com/stellar/typescript-wallet-sdk/pull/5/commits/f6b2de78f41959406599c3bcf7dc9a9e53c02f72